backend/src/org/commcare/xml/CommCareElementParser.java

```diff
@@ -46,18 +46,28 @@ public CommCareElementParser(KXmlParser parser) {
      * Build a DisplayUnit object by parsing the contents of a display tag.
      */
     public DisplayUnit parseDisplayBlock() throws InvalidStructureException, IOException, XmlPullParserException {
-        String imageValue = null;
-        String audioValue = null;
+        Text imageValue = null;
+        Text audioValue = null;
         Text displayText = null;
 
         while (nextTagInBlock("display")) {
             if (parser.getName().equals("text")) {
-                displayText = new TextParser(parser).parse();
+
+                String attributeValue = parser.getAttributeValue(null,"form");
+                
+                if(attributeValue != null && attributeValue.equals("image")){
+                    imageValue = new TextParser(parser).parse();
+                } else if(attributeValue != null && attributeValue.equals("audio")){
```

Comment:
Can you update the display element spec at https://github.com/dimagi/commcare/wiki/Suite20 ? It's hard to tell if the parser is doing the right thing here without being certain what kind of XML it's getting as input.